internal/storage/distlock_test.go

```diff
@@ -0,0 +1,95 @@
+package storage
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func storageProviders(t *testing.T) map[string]StorageProvider {
+	providers := map[string]StorageProvider{
+		"MEMORY": &MEMStorage{},
+	}
+
+	if (os.Getenv("ETCD_HOST") != "") && (os.Getenv("ETCD_PORT") != "") {
+		providers["ETCD"] = &ETCDStorage{}
+	}
+
+	if (os.Getenv("POSTGRES_HOST") != "") && (os.Getenv("POSTGRES_PORT") != "") {
+		providers["POSTGRES"] = &PostgresStorage{}
+	}
+
+	return providers
+}
+
+func distLockProviders(t *testing.T) map[string]DistributedLockProvider {
+	providers := map[string]DistributedLockProvider{
+		"MEMORY": &MEMLockProvider{},
+	}
+
+	if (os.Getenv("ETCD_HOST") != "") && (os.Getenv("ETCD_PORT") != "") {
+		providers["ETCD"] = &ETCDLockProvider{}
+	}
+
+	if (os.Getenv("POSTGRES_HOST") != "") && (os.Getenv("POSTGRES_PORT") != "") {
+		providers["POSTGRES"] = &PostgresLockProvider{}
+	}
+
+	return providers
+}
+
+func TestInit(t *testing.T) {
+	for name, dl := range distLockProviders(t) {
+		err := dl.Init(nil)
+		if err != nil {
+			t.Errorf("DistLock Init() failed: %v for provider: %s", err, name)
+		}
+	}
+}
+
+func TestInitFromStorage(t *testing.T) {
+	storageProviders := storageProviders(t)
+
+	for name, dl := range distLockProviders(t) {
+		ds := storageProviders[name]
+		// Doesn't return an error!
+		dl.InitFromStorage(ds, nil)
+	}
+}
```

Comment:
Even aside https://github.com/OpenCHAMI/power-control/issues/24 we should just discard this function from the interface. A weird partial copy that assumes traits about the input object is just silly.

We only use this as a convenience function in two tests. Those can just call `Init()` instead as a quick refactor.